### ISSUE TITLE
Modify FormData in FormFlow event and saveStepData after that

### ISSUE THREAD

**Issue by rvanlaak** [craue/CraueFormFlowBundle#113]
I've used all different types of events, but with none of them it is possible to use the `FormData` from the current request, do something with that data, and save the new data to the storage.

My use case is that I want to save a `UploadedFile` to the database, and store the id of that new Entity into the `FormFlow` by using `saveStepData()`. 

The problem is, that this data gets overwritten if the controller calls 
    $flow->saveCurrentStepData($form);

Since no event are triggered after the above call, the data from the storage always will get overwritten by the `formData`.

In other words: in what way is it possible to add extra info to the storage, after the form has been submitted and validated? Idea to dispatch two new events, called `PRE_STORAGE_DATA` and `POST_STORAGE_DATA`?
